File: solutions/javascript/difference-of-squares/1/difference-of-squares.js
//
// This is only a SKELETON file for the 'Difference Of Squares' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export class Squares {
  constructor(num) {
    this.numbers = [...Array(num).keys()].map(i => i + 1);
  }

  get sumOfSquares() {
    return this.numbers.reduce((acc, n)=> acc + n*n, 0);
  }

  get squareOfSum() {
    return (this.numbers.reduce((acc, n)=> acc + n, 0))**2;
  }

  get difference() {
    return this.squareOfSum - this.sumOfSquares;
  }
}

File: solutions/python/difference-of-squares/1/difference_of_squares.py
def square_of_sum(number):
    s = sum([n for n in range(number+1)])
    return s**2


def sum_of_squares(number):
    s = sum([n**2 for n in range(number+1)])
    return s


def difference_of_squares(number):
    return square_of_sum(number) - sum_of_squares(number)
